### PR TITLE
[TAS-231] ✨ Enable editing of edition info

### PR DESCRIPTION
**Pull request by AuroraHuang22**
https://github.com/likecoin/nft-book-press/assets/75730405/0602c0eb-91f4-4517-8e36-f5087b8048b0

### CODE REVIEW

**Comment by nwingt** [comment]
image

File: pages/nft-book-store/new.vue [hunk 1]
```diff
@@ -252,23 +304,25 @@ function sanitizeHtml (html: string) {
 
 async function onSubmit () {
```
Comment (nwingt) [inline pages/nft-book-store/new.vue:305]: How about separating `onSubmit()` into 2 sub-functions for new and edit
e.g. `submitNewClass()`, `submitEditedClass()`

```suggestion
async function onSubmit () {
  return this.isEditMode.value ? this.submitEditedClass() : this.submitNewClass();
}
```

File: pages/nft-book-store/new.vue [hunk 2]
```diff
@@ -316,6 +379,31 @@ async function onSubmit () {
   }
 }
 
+async function submitEditedClass () {
+  try {
+    isLoading.value = true
+
+    const p = mapPrices(prices.value)
+    await checkStripeConnect()
```
Comment (williamchong) [inline pages/nft-book-store/new.vue:387]: Why check stripe connect here?

File: pages/nft-book-store/new.vue [hunk 3]
```diff
@@ -316,6 +379,31 @@ async function onSubmit () {
   }
 }
 
+async function submitEditedClass () {
+  try {
+    isLoading.value = true
```
Comment (williamchong) [inline pages/nft-book-store/new.vue:384]: Not sure if we need to guard if we are really in edit mode, and `editionIndex` is properly set here

**Review by williamchong — APPROVED** [review]

**Review by nwingt — APPROVED** [review]

**Pull request merged**